### PR TITLE
Add warnings about missing dom ext and require json ext

### PR DESCRIPTION
**Pull request by keradus**

### CODE REVIEW

File: src/Report/JunitReporter.php [hunk 1]
```diff
@@ -34,6 +34,10 @@ public function getFormat()
      */
     public function generate(ReportSummary $reportSummary)
     {
+        if (!extension_loaded('dom')) {
```
Comment (SpacePossum) [inline src/Report/JunitReporter.php]: Can we check this when parsing the config? Than the user will know the report cannot be generated before fixing, maybe better UX?

Comment (keradus) [inline src/Report/JunitReporter.php]: early check would be nice indeed, but since this is a public method, we need to check it here nevertheless

Comment (keradus) [inline src/Report/JunitReporter.php]: let us keep this PR as a quickfix, adding nice ahead-of time warning would be great improvement, but it shall deserve a dedicated PR to provide it in nice version (instead of version provided fast)
